test/master-playlist-controller.test.js

```diff
@@ -872,6 +877,157 @@ QUnit.test('re-triggers bandwidthupdate events on the tech', function(assert) {
   assert.equal(bandwidthupdateEvents, 2, 'triggered bandwidthupdate');
 });
 
+QUnit.test('switches to lower renditions immediately, higher dependent on buffer',
+function(assert) {
+  this.masterPlaylistController.mediaSource.trigger('sourceopen');
+  // master
+  this.standardXHRResponse(this.requests.shift());
+  // media
+  this.standardXHRResponse(this.requests.shift());
+
+  let buffered = [];
+  let currentPlaylistBandwidth = 0;
+  let nextPlaylistBandwidth = 0;
+  let mediaChanges = [];
+  let currentTime = 0;
+  let endList = true;
+
+  this.masterPlaylistController.tech_.currentTime = () => currentTime;
+  this.masterPlaylistController.tech_.buffered = () => videojs.createTimeRanges(buffered);
+  this.masterPlaylistController.selectPlaylist = () => {
+    return {
+      attributes: {
+        BANDWIDTH: nextPlaylistBandwidth
+      },
+      endList
+    };
+  };
+  this.masterPlaylistController.masterPlaylistLoader_.media = (media) => {
+    if (!media) {
+      return {
+        attributes: {
+          BANDWIDTH: currentPlaylistBandwidth
+        },
+        endList
+      };
+    }
+    mediaChanges.push(media);
+  };
+
+  currentTime = 0;
+  currentPlaylistBandwidth = 1000;
+  nextPlaylistBandwidth = 1000;
+  buffered = [];
+  this.masterPlaylistController.mainSegmentLoader_.trigger('bandwidthupdate');
+  assert.equal(mediaChanges.length,
+               1,
+               'changes media when no buffer and equal bandwidth playlist');
```

Comment:
It looks like it succeeds here because `currentTime === 0` and initial `BUFFER_LOW_WATER_LINE` is 0, so the check `forwardBuffer >= bufferLowWaterLine` is `true`.